rs/nervous_system/tools/sync-with-released-nervous-system-wasms/src/main.rs

```diff
@@ -52,6 +69,95 @@ async fn get_mainnet_canister_git_commit_id_and_module_hash(
     Ok((git_commit_id, module_hash))
 }
 
+async fn get_mainnet_canister_git_tag_and_module_hash(
+    agent: &Agent,
+    canister_id: CanisterId,
+    canister_repository: String,
+    canister_name: String,
+) -> Result<(String, String)> {
+    use std::fmt::Write;
+
+    #[derive(Debug, Deserialize)]
+    struct Tag {
+        name: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    struct ReleaseAsset {
+        name: String,
+        browser_download_url: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    struct Release {
+        tag_name: String,
+        assets: Vec<ReleaseAsset>,
+    }
+
+    const GITHUB_API: &str = "https://api.github.com";
+
+    let canister_id = canister_id.get().0;
+
+    let module_hash = agent
+        .read_state_canister_info(canister_id, "module_hash")
+        .await?;
+    let module_hash = module_hash.iter().fold(String::new(), |mut output, x| {
+        let _ = write!(output, "{:02x}", x);
+        output
+    });
+
+    let token = std::env::var("GITHUB_TOKEN").expect("Set GITHUB_TOKEN env var");
+
+    let client = Client::builder()
+        .user_agent("sync-with-released-nervous-system-wasms")
+        .build()?;
+
+    let tags_url = format!("{}/repos/{}/tags", GITHUB_API, canister_repository);
+    let tags: Vec<Tag> = client
+        .get(&tags_url)
+        .bearer_auth(&token)
+        .send()
+        .await?
+        .json()
+        .await?;
+
+    for tag in tags.iter() {
+        let release_url = format!(
+            "{}/repos/{}/releases/tags/{}",
+            GITHUB_API, canister_repository, tag.name
+        );
+        let res = client.get(&release_url).bearer_auth(&token).send().await?;
+
+        if res.status().is_success() {
+            let release: Release = res.json().await?;
+
+            let canister_sha256_name = format!("{}.sha256", canister_name);
+            let canister_sha256_url = release
+                .assets
+                .iter()
+                .find(|asset| asset.name == canister_sha256_name)
+                .map(|asset| asset.browser_download_url.clone());
+            if let Some(canister_sha256_url) = canister_sha256_url {
+                let canister_sha256 = client
+                    .get(canister_sha256_url)
+                    .bearer_auth(&token)
+                    .send()
+                    .await?
+                    .text()
+                    .await?;
+                if canister_sha256.starts_with(&module_hash) {
```

Comment:
Let's ensure above that `module_hash` is indeed 256 bits, otherwise, this condition could boil down to `true` if `module_hash` is somehow `""`.